src/components/dropdownSearch/MultiSelectDropdownSearch/MultiSelectDropdownSearchReducer.tsx

```diff
@@ -0,0 +1,134 @@
+import { DropdownSearchActions, IOptionsDropdownSearchPayload } from '../DropdownSearchActions';
+import {
+  addUniqueSelectedOption,
+  dropdownSearchInitialState,
+  dropdownSearchReducer,
+  getDisplayedOptions,
+  getNextIndexPosition,
+  getOptionsFiltered,
+  IDropdownSearchState,
+  removeSelectedOption,
+} from '../DropdownSearchReducers';
+import { IReduxAction } from '../../../utils/ReduxUtils';
+import {FixedQueue} from '../../../utils/FixedQueue';
+import {IDropdownOption} from '../DropdownSearch';
+import {keyCode} from '../../../utils/InputUtils';
+
+export const multiSelectDropdownSearchReducer = (state: IDropdownSearchState = dropdownSearchInitialState,
+  action: IReduxAction<IOptionsDropdownSearchPayload>) => {
+
+  switch (action.type) {
+    case DropdownSearchActions.addMultiSelect:
+      return {
+        ...state,
+        id: action.payload.id,
+        options: action.payload.optionsDropdown,
+        selectedOptions: new FixedQueue<IDropdownOption>(),
+        filterText: action.payload.filterText,
+        isOpened: false,
+      };
+    case DropdownSearchActions.removeSelectedOption:
+      state.selectedOptions = removeSelectedOption(state, action.payload.selectedOptionValue);
+      state.displayedOptions = getDisplayedOptions(state);
+      return {
+        ...state,
+        id: action.payload.id,
+        selectedOptions: state.selectedOptions,
+        isOpened: false,
+        activeOption: undefined,
+        displayedOptions: state.displayedOptions,
+      };
+    case DropdownSearchActions.removeAllSelectedOptions:
+      state.selectedOptions = new FixedQueue<IDropdownOption>();
+      state.displayedOptions = getDisplayedOptions(state);
+      return {
+        ...state,
+        id: action.payload.id,
+        selectedOptions: state.selectedOptions,
+        displayedOptions: state.displayedOptions,
+      };
+    case DropdownSearchActions.multiSelect:
+      return {
+        ...state,
+        id: action.payload.id,
+        isOpened: true,
+        selectedOptions: state.selectedOptions.push(action.payload.addedSelectedOption),
+        displayedOptions: getDisplayedOptions(state),
+      };
+    case DropdownSearchActions.addCustomSelectedOption:
+      return {
+        ...state,
+        id: action.payload.id,
+        isOpened: true,
+        selectedOptions: addUniqueSelectedOption(state, action.payload.selectedOptionValue),
+        displayedOptions: getDisplayedOptions(state),
+      };
+    case DropdownSearchActions.onKeyDownMultiselect:
```

Comment:
KeyCode must be in the payload otherwise how can we know what to do when the user presses a key? The other way would be to dispatch a specific action on a specific key but this would be handled in each component on key. The logic would either be in the reducer or in the component. We decided to put it in the reducer, but is it the wisest choice? I might go read on that.